R/module_data_summary.R

```diff
@@ -3,15 +3,17 @@
 #' Module and its utils to display the number of rows and subjects in the filtered and unfiltered data.
 #'
 #' @details Handling different data classes:
-#' `get_object_filter_overview()` is a pseudo S3 method which has variants for:
+#' `get_filter_overview()` is a pseudo S3 method which has variants for:
```

Comment:
In the future it can be easily changed into S3 - so that app developers/teal-developers can make own methods to display summary for other datasets. _There is a vision for `teal.slice` that it could be extended in the future by S3 methods._

Reply:
Can we make it S3 without exporting the method?

Reply:
We can but I wouldn't do this now. This won't even change the code dramatically because we can't export `<method>.data.frame` or `<method>.MultiAssayExperiment` because it will complicate overriding in the future by the users. When trying to refactor `teal.slice` (you were also involved i remember), we noticed that overriding a S3 method is possible only through `.GlobalEnv` - if somebody creates `<method>.data.frame` in the package then `teal` will always prioritize `teal` method. This is why I'd avoid creating S3 for now, because the only method will be `get_filter_overview.default` with this "ugly-if--pseudo-dispatch"